mailbox/opensearch/src/main/java/org/apache/james/mailbox/opensearch/events/OpenSearchListeningMessageSearchIndex.java

```diff
@@ -173,11 +179,15 @@ private Mono<Void> handleMailboxEvent(Event event, MailboxSession session, Mailb
     }
 
     private Mono<Void> handleExpungedEvent(MailboxEvents.Expunged expunged, MailboxSession session, MailboxId mailboxId) {
-        if (expunged.isMoved()) {
-            return Mono.justOrEmpty(expunged.movedToMailboxId())
-                .flatMap(movedToMailboxId -> Flux.fromIterable(expunged.getExpunged().values())
-                    .concatMap(oldMessageData -> handleExpungedMessage(oldMessageData, mailboxId, movedToMailboxId, session))
-                    .then(Mono.defer(() -> delete(session, mailboxId, expunged.getUids()))));
+        if (messageIndexNewBehavior) {
+            if (expunged.isMoved()) {
```

Comment:
`if (messageIndexNewBehavior && expunged.isModed())` ?

Then you don't need the nested if

Reply:
I thought about it first, But I think the separate if will be more "readable"

Reply:
I still feel the two same deletes afterwards being redundant... Or actually better, remove the else clauses and only keep a return delete statement at the end of the function? :)